components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/processor/JsonRestProcessor.java

```diff
@@ -104,6 +108,11 @@ protected void processRequest(Exchange exchange) throws SalesforceException {
             });
             break;
 
+        case RECENT:
+            // handle known response type
```

Comment:
Ah yeah other way around is better. You are welcome to fix that too. Just that its consistent, so next time someone add something he/she does the same.